File: CODE/DART/lib/writing.dart
// -- IMPORTS

import "dart:io";
import "building.dart";

// -- FUNCTIONS

void writeFileText(
    String filePath,
    String fileText
    )
{
    File( filePath ).writeAsStringSync( fileText );
}

// ~~

void writeGsonValue(
    String filePath,
    dynamic value,
    [
        void Function( String, String )? writeFileTextFunction = writeFileText
    ]
    )
{
    writeFileTextFunction?.call( filePath, buildGsonText( value ) );
}

File: CODE/PHP/writing.php
<?php
// -- IMPORTS

require_once __DIR__ . '/building.php';

// -- FUNCTIONS

function writeFileText(
    $filePath,
    $fileText,
    $ignored = null
    )
{
    file_put_contents( $filePath, $fileText );
}

// ~~

function writeGsonValue(
    $filePath,
    $value
    )
{
    writeFileText( $filePath, buildGsonText( $value ) );
}
